presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1183,6 +1185,21 @@ private PhysicalOperation visitScanFilterAndProject(
                 sourceLayout = source.getLayout();
             }
 
+            // filterExpression may contain large function calls; evaluate them before compiling.
+            if (filterExpression.isPresent()) {
+                Type type = filterExpression.get().getType();
+                Object value = new RowExpressionInterpreter(filterExpression.get(), metadata, session, true).optimize();
+                if (value instanceof RowExpression) {
+                    RowExpression optimizedFilter = (RowExpression) value;
+                    // building channel info
+                    optimizedFilter = SymbolToChannelTranslator.translate(optimizedFilter, sourceLayout);
+                    filterExpression = Optional.of(optimizedFilter);
+                }
+                else {
+                    filterExpression = Optional.of(constant(value, type));
```

Comment:
Here, we will either discard the all rows or keep all rows?

Reply:
What do you mean by that?

Reply:
FilterExpression will only have TRUE or FALSE since it is a boolean expression

Reply:
That is an interesting point! I will add a TODO

Reply:
For this kind of TODO, i feel that we should create issues. Otherwise it will still be a TODO 5 years from now, like all the TODOs in the code from 5 years ago...

Reply:
Created #12660